transcom/mymove
# INT - B-21412 & B-21413 Prime sim - create international crate - KICKBACK

## [B-21412](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21412) & [B-21413](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21413)

INT PR: https://github.com/transcom/mymove/pull/14223

## Summary

Found a bug where a boolean check is compared incorrectly. Fixed to compare `address.isOconus` correctly to allow it to populate the correct value for `market` field.

### How to test

1. In your `.envrc` file -> change to `export FEATURE_FLAG_ENABLE_ALASKA=true` to turn on Alaska FF locally.
2. Close your server -> run `direnv allow` and then restart your server.
3. Use `HHGMoveInSITEndsTomorrow` testharness and find the move as TOO.
4. Edit shipment -> and choose `AK` as the state for destination/delivery location and click save. (The move should be iHHG now).
5. In Prime Sim -> find the move.
6. Click `Add Service Item` button for the shipment.
7. Select `International Crating` for "Service item type".
8. Select `International Crating` for "Service item code".
10. Make sure item length/width/height is less than the Crate length/width/height.
11. Open up browser dev tools or inspect tools.
12. After clicking "Create service item" and is successful -> check the `mto-service-item` POST api returned value and verify that field shows "CONUS". (You need to check the returned value from the POST and not from GET.)
13. Repeat steps 5 to 12 but with "International Uncrating" and you should see "OCONUS" for `market`.


## Screenshots
![image](https://github.com/user-attachments/assets/c2ad746c-8234-428a-aace-c756acda0143)
